nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -375,13 +482,16 @@ in {
         { assertion = !(acfg.dbpass != null && acfg.dbpassFile != null);
           message = "Please specify no more than one of dbpass or dbpassFile";
         }
-        { assertion = ((acfg.adminpass != null || acfg.adminpassFile != null)
-            && !(acfg.adminpass != null && acfg.adminpassFile != null));
+        { assertion = (acfg.adminpass != null) != (acfg.adminpassFile != null);
           message = "Please specify exactly one of adminpass or adminpassFile";
         }
         { assertion = versionOlder cfg.package.version "21" -> cfg.config.defaultPhoneRegion == null;
           message = "The `defaultPhoneRegion'-setting is only supported for Nextcloud >=21!";
         }
+        { assertion = acfg.objectstore == null
+            || (lists.count (v: v.enable) (attrsets.attrValues acfg.objectstore)) == 1;
+          message = "If using objectstore class as primary storage exactly one class can be enabled.";
+        }
```

Comment:
With my suggestion from above we can drop this assertion entirely.

Reply:
This check will become useful in the case that options are added for the other nextcloud-supported classes of object storage as mentioned above, as according to the nextcloud docs a user should only be allowed to select a single object storage class to use as primary storage.

Reply:
This feels like premature optimization. As suggested above, `objectstore.s3.<option>` is good-enough for a tentative design, more things (including such an assertion) can be added later when we need it.